src/averageAges.js

```diff
@@ -15,11 +15,20 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredMale = century
+    ? people.filter(
+      (person) =>
+        Math.ceil(person.died / 100) === century && person.sex === 'm'
+    )
+    : people.filter(
+      (person) => person.sex === 'm'
+    );
+
+  const maleAge = filteredMale.map((person) => person.died - person.born);
+  const totalAge = maleAge.reduce((total, num) => total + num, 0);
+  const average = totalAge / filteredMale.length;
+
```

Comment:
you have the same calculations in `calculateWomenAverageAge` function. please create separate function with param and reuse it